racksdb/web/ui/src/views/InfrastructureDetailsView.vue

```diff
@@ -0,0 +1,161 @@
+
+const http = useHttp()
+const infrastructures: Ref<Array<Infrastructure>> = ref([])
+const infrastructureDetails: Ref<Infrastructure | undefined> = ref()
+const showFullImg = ref(false)
+const rack = ref()
+const cardsView = ref(true)
+const tableView = ref(false)
+const showRack = ref(false)
+const showInfrastructureRacks = ref(true)
+
+function choseView(){
+    if(cardsView.value){
+        cardsView.value = !cardsView.value
+        tableView.value = !tableView.value 
+    }else{
+        tableView.value = !tableView.value
+        cardsView.value = !cardsView.value
+    }
+}
+
+function openImg(){
+    showFullImg.value = !showFullImg.value
+}
+
+function rackDetails(rackName: string){
+    showRack.value = true
+    showInfrastructureRacks.value = false
+    var layout = infrastructureDetails.value?.layout
+
+    for (let index = 0; index < layout!.length; index++) {
+        rack.value = layout!.filter(rack => rack.rack === rackName)
+    }
+}
+
+async function getInfrastructure(){
+    try {
+        const resp = await http.get('infrastructures')
+        infrastructures.value = resp.data as Infrastructure[]
+        infrastructureDetails.value = infrastructures.value.filter((infrastructure) => infrastructure.name === props.name)[0]
+    } catch (error) {
+        console.error('Error during infrastructure data recovery', error)
+    }
+}
+
+onMounted(() => {
+    getInfrastructure()
+})
+
+const props = defineProps({
+    name: String
+})
+</script>
+
+<template>
+    <SearchBar 
+        v-if="infrastructures.length"
+            viewTitle="Infrastructure Details"    
+            searchedItem="infrastructure" 
+            :items="infrastructures"/>
+
+    <h2 class="text-3xl font-medium flex justify-center capitalize py-16">{{ name }} Infrastructure</h2>
+
+    <img :src="`${inject(injectionKey)!.api_server}/draw/infrastructure/${ props.name }.svg`" alt="" @click="openImg()" class="h-96 max-w-500 mx-auto p-10">
+    
+    <div v-show="showFullImg" class="fixed top-0 left-0 flex flex-col items-center justify-center w-screen h-screen bg-gray-300 bg-opacity-50 dark:bg-gray-900 dark:bg-opacity-50 backdrop-blur-md z-50">
+        <img :src="`${inject(injectionKey)!.api_server}/draw/infrastructure/${ props.name }.svg`" alt="" @click="openImg()" class="h-screen max-w-full bg-white">
+    </div>
+    
+    <div class="flex justify-end mr-36">
+        <img @click="choseView()" src="/assets/cards.svg" alt="">
+        <img @click="choseView()" src="/assets/table.svg" alt="">
+    </div>
+
+    <div v-show="cardsView">
+        <InfrastructureCards v-if="infrastructureDetails" :infrastructure="infrastructureDetails" searchItem="nodes" />
+        <InfrastructureCards v-if="infrastructureDetails" :infrastructure="infrastructureDetails" searchItem="storage" />
+        <InfrastructureCards v-if="infrastructureDetails" :infrastructure="infrastructureDetails" searchItem="network" />
+    </div>
+
+    <div v-show="tableView" class="pb-10">
```

Comment:
Cards are in a component and the table is not. Please create an infrastructure table component to be consistent for both ways of representing the data.